### PR TITLE
fix(my-account): fixes for My Account

### PR DESCRIPTION
**Pull request by dkoo**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-theme/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Companion to https://github.com/Automattic/newspack-plugin/pull/2904.

### How to test the changes in this Pull Request:

Follow instructions in https://github.com/Automattic/newspack-plugin/pull/2904.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

**Comment by dkoo** [comment]
Note: Rebased against `epic/ras-acc` as gift subscription functionality has been fixed in the epic, not in `trunk`.

File: includes/class-modal-checkout.php [hunk 1]
```diff
@@ -78,69 +75,6 @@ public static function init() {
 		add_filter( 'googlesitekit_tagmanager_tag_blocked', [ __CLASS__, 'is_modal_checkout' ] );
 		add_filter( 'jetpack_active_modules', [ __CLASS__, 'jetpack_active_modules' ] );
 	}
-
-	/**
-	 * Add information about modal checkout to cart item data.
-	 *
-	 * @param array $cart_item_data Cart item data.
-	 */
-	public static function amend_cart_item_data( $cart_item_data ) {
```
Comment (chickenn00dle) [inline includes/class-modal-checkout.php:87]: I'm getting the following error when testing with https://github.com/Automattic/newspack-blocks/pull/1661, https://github.com/Automattic/newspack-plugin/pull/2904, and https://github.com/Automattic/newspack-theme/pull/2240 checked out:

```
[06-Feb-2024 16:14:03 UTC] PHP Fatal error:  Uncaught Error: Call to undefined method Newspack_Blocks\Modal_Checkout::amend_cart_item_data() in /newspack-repos/newspack-blocks/includes/class-modal-checkout.php:155
Stack trace:
#0 /var/www/html/wp-includes/class-wp-hook.php(324): Newspack_Blocks\Modal_Checkout::process_checkout_request()
#1 /var/www/html/wp-includes/class-wp-hook.php(348): WP_Hook->apply_filters()
#2 /var/www/html/wp-includes/plugin.php(517): WP_Hook->do_action()
#3 /var/www/html/wp-settings.php(665): do_action()
#4 /var/www/html/wp-config.php(115): require_once('...')
#5 /var/www/html/wp-load.php(50): require_once('...')
#6 /var/www/html/wp-blog-header.php(13): require_once('...')
#7 /var/www/html/index.php(17): require('...')
#8 {main}
  thrown in /newspack-repos/newspack-blocks/includes/class-modal-checkout.php on line 155
```

Seems this method is still needed here: https://github.com/Automattic/newspack-blocks/blob/9af158260a287b45328e43abead1ae5d29628b87/includes/class-modal-checkout.php#L155

**Review by chickenn00dle — CHANGES_REQUESTED** [review]

File: includes/class-modal-checkout.php (hunk 1 above)
Comment (dkoo) [reply includes/class-modal-checkout.php:87]: Ugh, thanks. Trying to multitask too much 🤦

Comment (dkoo) [reply includes/class-modal-checkout.php]: Fixed in 12da73a

**Review by chickenn00dle — APPROVED** [review]
Nice! Thanks for the quick update. This lgtm 👍

**Pull request merged**